pkg/skaffold/build/custom/custom.go

```diff
@@ -52,19 +56,61 @@ func NewArtifactBuilder(pushImages bool, additionalEnv []string) *ArtifactBuilde
 // Build builds a custom artifact
 // It returns true if the image is expected to exist remotely, or false if it is expected to exist locally
 func (b *ArtifactBuilder) Build(ctx context.Context, out io.Writer, a *latest.Artifact, tag string) error {
-	cmd, err := b.retrieveCmd(ctx, out, a, tag)
+	cmd, err := b.retrieveCmd(out, a, tag)
 	if err != nil {
 		return errors.Wrap(err, "retrieving cmd")
 	}
 
-	return cmd.Run()
+	if err := cmd.Start(); err != nil {
+		return errors.Wrap(err, "starting cmd")
+	}
+
+	done := make(chan struct{})
+	go func() {
+		select {
```

Comment:
This code block is really long. Can we refactor this in to `handleGracefulTermination`